lib/restforce/concerns/api.rb

```diff
@@ -383,7 +385,7 @@ def upsert!(sobject, field, attrs)
                       "#{ERB::Util.url_encode(external_id)}", attrs
           end
 
-        response.body.respond_to?(:fetch) ? response.body.fetch('id', true) : true
+        response.body.slice("id", "created")
```

Comment:
There are some Salesforce API versions where this API returns `204 No Content` after an update ([source](https://developer.salesforce.com/docs/atlas.en-us.api_rest.meta/api_rest/dome_upsert.htm)):

> In API version 45.0 and earlier, the HTTP status code is 204 (No Content) and there isn’t a response body.

I think using `#slice` will break for those API versions 😦

Reply:
I see :/ To be honest, I'm not very familiar with Salesforce, even less with old versions of it.

Maybe a solution could be:

```rb
# Update response for old API implementations that aren't returning the id
if !response.body.respond_to?(:fetch) || !response.body.has_key?("id")
  return { "created" => false }
end
# Create response for old API implementations that aren't returning "created"
if !response.body.has_key?("created")
  return { "id" => response.body["id"], "created" => true }
end
# New API
response.body.slice("id", "created")
```

But it becomes a bit complicated ^^"

Another possibility is to stay with the same Restforce spec (return true in case of Update, id in case of Create), like this:

```rb
# Update case for old API returning nothing
return true if !response.body.respond_to?(:fetch)
# Update case for old API returning "true"
return true if !response.body.has_key?("id")
# Update case for new API returning "created"
return true if response.body.has_key?("created") && response.body["created"] == false
# Create case
response.body["id"]
```

This last solution would solve https://github.com/restforce/restforce/issues/869 and https://github.com/restforce/restforce/issues/556 but not https://github.com/restforce/restforce/issues/253